tests/unit/test_summary.py

```diff
@@ -214,9 +215,17 @@ def test_dictify_converts_facet_count_to_dictionary():
 
 
 def test_split_author_turns_string_into_compound_field():
-    assert split_author('1234:Foo Bar') == {'mitid': 1234, 'name': 'Foo Bar'}
+    assert split_author('1234:Foo Bar') == {'mitid': '1234', 'name': 'Foo Bar'}
 
 
 def test_split_author_returns_none_for_invalid_author():
     assert split_author(':Foo Bar') is None
     assert split_author('Foo Bar') is None
+
+
+def test_join_author_joins_author_parts():
+    assert join_author({'name': 'Cat, Lucy', 'mitid': '1234'}) == \
+        '1234:Cat, Lucy'
+
+def test_join_author_uses_empty_string_for_missing_items():
```

Comment:
In the spirit of completeness, probably, but the data should never be such that we have an ID but no name, or neither a name nor an ID.